From 580bff5a09f1dcc45aac6edd178b272c50e95187 Mon Sep 17 00:00:00 2001
From: Myron Stowe <myron.stowe@redhat.com>
Date: Fri, 5 Sep 2014 01:31:51 -0400
Subject: [iommu] amd: Use dev_is_pci() to check whether it is pci device

Message-id: <20140905013151.30867.91870.stgit@gir.stowe>
Patchwork-id: 91271
O-Subject: [RHEL7.1 PATCH 07/28] iommu/amd: Use dev_is_pci() to check whether it is pci device
Bugzilla: 1129808
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Don Dutile <ddutile@redhat.com>
RH-Acked-by: Ivan Vecera <ivecera@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Alex Williamson <alex.williamson@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1129808
Upstream Status: b82a2272b37af1f1f86ee6e5966ad941f9db5dc7

commit b82a2272b37af1f1f86ee6e5966ad941f9db5dc7
Author: Yijing Wang <wangyijing@huawei.com>
Date:   Thu Dec 5 19:42:41 2013 +0800

    iommu/amd: Use dev_is_pci() to check whether it is pci device

    Use PCI standard marco dev_is_pci() instead of directly compare
    pci_bus_type to check whether it is pci device.

    Signed-off-by: Yijing Wang <wangyijing@huawei.com>
    Signed-off-by: Joerg Roedel <joro@8bytes.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/iommu/amd_iommu.c b/drivers/iommu/amd_iommu.c
index 76e579a..71776ff 100644
--- a/drivers/iommu/amd_iommu.c
+++ b/drivers/iommu/amd_iommu.c
@@ -248,8 +248,8 @@ static bool check_device(struct device *dev)
  if (!dev || !dev->dma_mask)
   return false;
 
- /* No device or no PCI device */
- if (dev->bus != &pci_bus_type)
+ /* No PCI device */
+ if (!dev_is_pci(dev))
   return false;
 
  devid = get_device_id(dev);
-- 
1.7.1